File: python/scrabble-score/scrabble_score.py
'''Given a word, compute the scrabble score for that word.

Letter                           Value
A, E, I, O, U, L, N, R, S, T       1
D, G                               2
B, C, M, P                         3
F, H, V, W, Y                      4
K                                  5
J, X                               8
Q, Z                               10
'''


def score(word: str) -> int:
    ''' Returns the scrabble score for a given word.'''
    points = 0
    for ch in word.upper():
        if ch in 'AEIOULNRST':
            points += 1
        elif ch in 'DG':
            points += 2
        elif ch in 'BCMP':
            points += 3
        elif ch in 'FHVWY':
            points += 4
        elif ch in 'K':
            points += 5
        elif ch in 'JX':
            points += 8
        elif ch in 'QZ':
            points += 10
    return points

File: go/scrabble-score/scrabble_score.go
// Package scrablle contains a routine that computes the scabble score of a string.
package scrabble

var table map[rune]int

func init() {
	table = map[rune]int{}
	for _, ch := range "AEIOULNRSTaeioulnrst" {
		table[ch] = 1
	}
	for _, ch := range "DGdg" {
		table[ch] = 2
	}
	for _, ch := range "BCMPbcmp" {
		table[ch] = 3
	}
	for _, ch := range "FHVWYfhvwy" {
		table[ch] = 4
	}
	for _, ch := range "Kk" {
		table[ch] = 5
	}
	for _, ch := range "JXjx" {
		table[ch] = 8
	}
	for _, ch := range "QZqz" {
		table[ch] = 10
	}
}

// Given a word, compute the Scrabble score for that word.
func Score(s string) int {
	score := 0
	for _, ch := range s {
		if sc, ok := table[ch]; ok {
			score += sc
		}
	}
	return score
}
